### ISSUE TITLE
prometheus-operator v0.52.1に更新後、svc/prometheus-k8sのEndpointsが<none>になった

### ISSUE THREAD

**Issue by oke-py** [cloudnativedaysjp/dreamkast-infra#1230]
ref #1223

#1229 でdevのprometheus-operatorをv0.48.1からv0.52.1に更新した。
その結果、svc/prometheus-k8sのEndpointsが<none>となり、kubectl port-forwardやGrafanaによるメトリクス表示ができなくなった。

```
kubectl -n monitoring get po prometheus-k8s-0 -o jsonpath='{.metadata.labels}' | jq .
```
v0.48.1 prd
```json
{
  "app": "prometheus",
  "app.kubernetes.io/component": "prometheus",
  "app.kubernetes.io/instance": "k8s",
  "app.kubernetes.io/managed-by": "prometheus-operator",
  "app.kubernetes.io/name": "prometheus",
  "app.kubernetes.io/part-of": "kube-prometheus",
  "app.kubernetes.io/version": "2.26.0",
  "controller-revision-hash": "prometheus-k8s-64c45dc456",
  "operator.prometheus.io/name": "k8s",
  "operator.prometheus.io/shard": "0",
  "prometheus": "k8s",
  "statefulset.kubernetes.io/pod-name": "prometheus-k8s-0"
}
```
v0.52.1 dev
```json
{
  "app.kubernetes.io/component": "prometheus",
  "app.kubernetes.io/instance": "k8s",
  "app.kubernetes.io/managed-by": "prometheus-operator",
  "app.kubernetes.io/name": "prometheus",
  "app.kubernetes.io/part-of": "kube-prometheus",
  "app.kubernetes.io/version": "2.26.0",
  "controller-revision-hash": "prometheus-k8s-79595dfd46",
  "operator.prometheus.io/name": "k8s",
  "operator.prometheus.io/shard": "0",
  "prometheus": "k8s",
  "statefulset.kubernetes.io/pod-name": "prometheus-k8s-0"
}
```

ここを削除すればよさそう。
https://github.com/cloudnativedaysjp/dreamkast-infra/blob/5fd56341140f0c977ff34ff1e8ae156f84657855/manifests/infra/prometheus/base/prometheus.yaml#L181

**Comment by mayocream** [comment]
Since 0.50, prometheus-operator removes "app" label selector: https://github.com/prometheus-operator/prometheus-operator/pull/4350

kube-prometheus also have a commit related to it: [remove "app" label selector deprecated by prometheus-operator](https://github.com/prometheus-operator/kube-prometheus/commit/b6c97fc6c03dea43f888f403566ae04ecde9e9a5#diff-7d96cb4d077b331c98e172f727d62a7d4f626236fdc70f27548b7459c29a11b8), this removed `app: prometheus` from the prometheus svc.

We need to sync changes from [kube-prometheus manifests](https://github.com/prometheus-operator/kube-prometheus/tree/main/manifests).

**Comment by mayocream** [comment]
Can we use kustomize like this? @oke-py 
```
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization
namespace: monitoring
resources:
- https://raw.githubusercontent.com/prometheus-operator/kube-prometheus/612a79fda277134f51390e555b32846286f6ab40/manifests/prometheus-service.yaml
- prometheus.yaml
- node-exporter.yaml
- monitor.yaml
```

[This change](https://github.com/prometheus-operator/kube-prometheus/commit/b6c97fc6c03dea43f888f403566ae04ecde9e9a5#diff-7d96cb4d077b331c98e172f727d62a7d4f626236fdc70f27548b7459c29a11b8) is not included in the 0.9.0 release of kube-prometheus, unable to use a version tag in the url.

**Comment by oke-py** [comment]
We only refer to kube-prometheus, don't use the manifest for now.
`kustomization.yaml` is good.
We need to modify [prometheus.yaml](https://github.com/cloudnativedaysjp/dreamkast-infra/blob/main/manifests/infra/prometheus/base/prometheus.yaml), which is managed by ourselves.
